### DISCUSSION TITLE
Test developing: several tests on one objects

### DISCUSSION THREAD

**Discussion by oleg68** [GrandOrgue/grandorgue#2015] [Developers]
@rousseldenis Working on #2012 I'd like to write a first test using the framework you have provided.

I want to test the AND functionality of a switch.

Now there are a source file ``model/GOSwitch.cpp`` and a test file ``tests/GOSwitchTest.cpp``.

GOSwitch has several functionalities and I'd like to add a separate test for each functionality. Is it possible to have several tests for one class in the current testing model?

**Comment by rousseldenis** [comment]
> Is it possible to have several tests for one class in the current testing model?

Of course, you can add as many 'assert' as you want.

**Reply by oleg68** [reply]
But if some assert fails, the test will break and all following asserts won't be checked.

I want the several tests would be independent and would continue even one of them failed. Is it possible with a single test class?

**Reply by rousseldenis** [reply]
I don't remember but I think they continue and log the exceptions in the result.

To be tested.

Either, you can create another class.

**Reply by rousseldenis** [reply]
IMHO, creating other test classes is not annoying as they could represent a technical/functional part of the application (and you can name the class with an oriented one).

**Reply by oleg68** [reply]
But the current naming convention allows to create only one test class for one source class.

I think it would be better to create an inner class for each tested functionality.

For example, if `GODrawsStop` is a source class with several functionalities, we would create

```
class GOTestDrawStop {
public:
  class SwitchConnection  : public GOCommonControllerTest {
    virtual void run();
  }

  class InternalStates : public GOCommonControllerTest {
    virtual void run();
  }
}
```

**Reply by rousseldenis** [reply]
Of course, the test framework is meant to be enhanced. Feel free to split logic if you find it more convenient.

IMHO, the bricks are there and programmer should be able to build the tests structure he seems the more convenient.
